### ISSUE TITLE
`laminarc start` output could point to job run (URL/id)

### ISSUE THREAD

**Issue by tsieprawski** [ohwgiles/laminar#33]
When triggering child jobs, as described by [1], the only returned information about child job's result is exit code ([2]). Having lots of parent and child jobs executed it's easy (at least for me) to lose track which child job "belongs" to the parent one. Thus I think it would be cool for `start` result to contain also any "pointer" to the child job.

Examples of what I have in mind:
- feedback with job ID only, probably as stdout/stderr. Below Example assumes that job is viewable at URL http://laminar.example/jobs/ajob/75 :
```bash
$ laminarc start ajob
75
$ echo $?
0
```
- feedback with job URL. This probably involves `laminard` to become self-conscious about its URL, and somehow it would need to became configurable - not sure whether this is cool approach?
```bash
$ laminarc start ajob
http://laminar.example/jobs/ajob/75
$ echo $?
0
```

[1] http://laminar.ohwgiles.documentup.com/#job-chains
[2] https://github.com/ohwgiles/laminar/blob/master/src/laminar.capnp#L26

**Comment by ohwgiles** [comment]
Yep this is an interesting idea. A child job can already be traced back to its parent as described under "Job chains", but you are right that there is no easy way to trace in the other direction.

So far I avoided having laminar know about its URL because it's not particularly maintainable when it has to be configured both in the webserver/DNS and in laminar iteslf. If there were other features that require laminar to know its own URL then perhaps I would consider this, but I think on its own this is not a strong enough reason. So I prefer your first approach.

One potential way of making it more user-friendly could be outputting something matchable like `:laminar:start:ajob:75:`, which could then be converted by the javascript in the web UI to a relative link to the child job. Or even perhaps an appropriate ANSI escape could be (re)used so that the output would not appear different from running the command manually. This would be similar to what is already present for converting ANSI colors into HTML and has the advantage that it should not appear differently when running it in a regular console. But I am a bit wary of doing custom processing on the console output of a job in javascript...what do you think?

**Comment by tsieprawski** [comment]
@ohwgiles 
So the first approach is fine. But it would be worth to note somewhere, that if URLs were to be introduced, then here would be a fine place to have them too. (:

`:laminar:start:ajob:75:` is cooler than "mine" `75` output alone, it's more scalable. In case more stuff would be delivered to callee of `laminarc start`, then it becomes easy to expand it. `:laminar:start:ajob:75:anything:moreanything` or similar. Just I can't think of a sensible example (child's output and how to escape it? or an real URL that callee can anyway build?), but just in case it's expandable.

Yeah, I don't specially like parsing job output anywhere. It's a place for lots of false positives, and developers would need to be very careful about what is stdouted. I've seen real-life jenkins jobs, that do `git diff` to previous commit at the very beginning, or stuff can be documented in README-s and printed then (especially by linters). And it is not so uncommon for various tools to use ANSI escapes already, although I believe some are unused by the world and could be silently used here.
Also looks to me that laminar aims to be very minimal, lots of typical CI stuff is reused by typical linux tools (remote jobs = `ssh`, child jobs = regular run of other job, notifications = actually any command line tool). It's cool, and I think it that just printing `pointer to child job` (whichever of `:laminar:start:ajob:75`, `75`, or http://laminar.example/jobs/ajob/75) is sufficient for now.

**Comment by ohwgiles** [comment]
Let’s just do `75` for now and think about extending it later. I probably won’t get a chance to implement this for a couple of weeks though

**Comment by ohwgiles** [comment]
@tsieprawski Is this solution ok?

**Comment by palica** [comment]
can this be done also for laminarc trigger ?

**Comment by ohwgiles** [comment]
@palica not directly. `laminarc trigger` just queues the job (and it might be queued a while), and it only gets a number when it pops off the queue and actually starts executing. In the "fire-and-forget" case which `trigger` was supposed to be, you don't want it to be stuck waiting for the job to get off the queue. One thing I could do is add another command that starts a job and blocks until it comes off the queue but doesn't wait until execution is finished - this command would be able to return a number. In that case the naming convention would probably need to be modified to something like `queue`, `start` and `run` instead of `trigger` and `start`.

What do you think?

**Comment by palica** [comment]
that would be nice.

**Comment by ohwgiles** [comment]
tracked in #36
